### ISSUE TITLE
hspec-core 2.4.0 fails to build example

### ISSUE THREAD

**Issue by jeking3** [hspec/hspec#306]
Using Ubuntu 14.04 LTS and when I build a project (in this case, Apache Thrift using cmake to build the haskell library with cabal), the following error occurs in hspec-core 2.4.0 which does not occur in hspec-core 2.3.2:

```
Building hspec-core-2.4.0...
Preprocessing library hspec-core-2.4.0...
[ 1 of 22] Compiling Paths_hspec_core ( dist/build/autogen/Paths_hspec_core.hs, dist/build/Paths_hspec_core.o )
[ 2 of 22] Compiling Test.Hspec.Core.Timer ( src/Test/Hspec/Core/Timer.hs, dist/build/Test/Hspec/Core/Timer.o )
[ 3 of 22] Compiling Data.Algorithm.Diff ( vendor/Data/Algorithm/Diff.hs, dist/build/Data/Algorithm/Diff.o )
[ 4 of 22] Compiling Test.Hspec.Core.Formatters.Diff ( src/Test/Hspec/Core/Formatters/Diff.hs, dist/build/Test/Hspec/Core/Formatters/Diff.o )
[ 5 of 22] Compiling Test.Hspec.Core.Compat ( src/Test/Hspec/Core/Compat.hs, dist/build/Test/Hspec/Core/Compat.o )
[ 6 of 22] Compiling Test.Hspec.Core.Util ( src/Test/Hspec/Core/Util.hs, dist/build/Test/Hspec/Core/Util.o )
[ 7 of 22] Compiling Test.Hspec.Core.FailureReport ( src/Test/Hspec/Core/FailureReport.hs, dist/build/Test/Hspec/Core/FailureReport.o )
[ 8 of 22] Compiling Test.Hspec.Core.QuickCheckUtil ( src/Test/Hspec/Core/QuickCheckUtil.hs, dist/build/Test/Hspec/Core/QuickCheckUtil.o )
[ 9 of 22] Compiling Test.Hspec.Core.Formatters.Free ( src/Test/Hspec/Core/Formatters/Free.hs, dist/build/Test/Hspec/Core/Formatters/Free.o )
[10 of 22] Compiling Test.Hspec.Core.Example ( src/Test/Hspec/Core/Example.hs, dist/build/Test/Hspec/Core/Example.o )
src/Test/Hspec/Core/Example.hs:156:45:
    Couldn't match type `[Char]' with `FailureReason'
    Expected type: FailureReason
      Actual type: String
    In the second argument of `Failure', namely `err'
    In the expression: Failure Nothing err
    In a case alternative:
        HUnit.HUnitFailure err -> Failure Nothing err
Failed to install hspec-core-2.4.0
cabal: Error: some packages failed to install:
hspec-2.4.0 depends on hspec-core-2.4.0 which failed to install.
hspec-core-2.4.0 failed during the building phase. The exception was:
ExitFailure 1
make[2]: *** [lib/hs/thrift_cabal.stamp] Error 1
make[1]: *** [lib/hs/CMakeFiles/haskell_library.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by sol** [comment]
Nice catch.  The issue happens with HUnit < 1.3.  Can you verify, what version of HUnit are you using?

**Comment by sol** [comment]
Fixed in `2.4.1`.
